Framework/Geometry/src/Instrument/Component.cpp

```diff
@@ -393,6 +387,152 @@ Quat Component::getRotation() const {
       return m_parent->getRotation() * m_rot;
   }
 }
+*/
+
+/// Helper for legacy access mode. Returns the index of the component.
+size_t Component::index() const {
+  return m_map->componentIndex(this->getComponentID());
+}
+
+bool Component::hasComponentInfo() const {
+  const IComponent *root = m_base;
+  while (auto parent = root->getBareParent())
+    root = parent;
+  auto instrument = dynamic_cast<const Instrument *>(root);
+  return m_map->hasComponentInfo(instrument);
+}
+
+/// Return the relative position to the parent Component
+Kernel::V3D Component::getRelativePos() const {
+  if (m_map) {
+
+    if (hasComponentInfo() && !m_map->componentInfo().isDetector(index())) {
```

Comment:
Well, it makes it much harder to review. Have you considered doing everything in the visitor right now instead of waiting for a follow up PR?